metaflow_extensions/torchrun/plugins/torchrun_libs/torch_distributed_setup.py

```diff
@@ -0,0 +1,33 @@
+
+import json
+
+from metaflow import current
+
+from .datastore import TorchrunDatastore, task_sync_barrier
+from .constants import NODE_STARTED_VAR
+
+class KeyPaths:
+    Node = lambda x: f"{NODE_STARTED_VAR}/node_{x}"
+
+
+def setup_torch_distributed_env(
+    all_nodes_started_timeout,
+    flow_datastore,
+    polling_frequency=0.1,
+):
+
+    """  
+    Wait for the all workers to write the keys in the datastore.
+    """
+
+    datastore = TorchrunDatastore(flow_datastore=flow_datastore, pathspec=current.pathspec)
+    key_push_path = KeyPaths.Node(current.parallel.node_index)
+    datastore.put(key_push_path, json.dumps({"started": True}))
+    _all_keys = [KeyPaths.Node(i) for i in range(current.parallel.num_nodes)]
+    _lock_args = {
+        'description': 'Waiting for all nodes to start',
```

Comment:
This description is used when the error message is raised. We can call it `Task timed out waiting for the other nodes. `